src/Faulkner/state_manager.py

```diff
@@ -55,11 +56,28 @@ def move_to_next_state(self):
             return
 
         # if yaw error below tolerance threshold, straight ahead, else fix yaw
-        if math.fabs(self.state_vars.err_yaw) <= self.params.yaw_precision:
-            self.set_action_state(2) # go straight ahead
-        else:
+        if math.fabs(self.state_vars.err_yaw) > self.params.yaw_precision and not self.is_oscillating():
             self.set_action_state(1) # fix yaw
+        else:
+            self.set_action_state(2) # go straight ahead
```

Comment:
If we are oscillating, do we risk going off path too much since our next action is straight ahead?